lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -1576,6 +1576,17 @@ public void setTrackInactiveTintList(@NonNull ColorStateList trackColor) {
     invalidate();
   }
 
+  @Override
+  public void setVisibility(int visibility) {
+    super.setVisibility(visibility);
+
+    if (visibility != VISIBLE) {
+      for (TooltipDrawable label : labels) {
+        ViewUtils.getContentViewOverlay(this).remove(label);
+      }
```

Comment:
Do we need to add the label back when the slider becomes visible?

Reply:
Actually setting the visibility back to `VISIBLE` calls `onDraw()` again which adds all the labels. When I added it in the same function it showed two labels (honestly I thought that wasn't supposed to happen since it uses the same object, if you know why it happened I'd love to know).

Reply:
Ah you are right. onDraw() will add or remove label according to the settings. Can you add a comment to state that when the slider is visible, the label will be added in onDraw()?

Regarding `label.isVisible()`, I think it's becauseTooltipDrawable doesn't really implement the logic to respect that setting. (Drawable's visibility settings are not come for free I think. I has to be implemented by the subclasses.)

Reply:
> Can you add a comment to state that when the slider is visible, the label will be added in onDraw()?

Sure, I'll do that.

> Regarding `label.isVisible()`, I think it's becauseTooltipDrawable doesn't really implement the logic to respect that setting. (Drawable's visibility settings are not come for free I think. I has to be implemented by the subclasses.)

Oh, I see. Thanks!